Makefile

```diff
@@ -9,6 +9,7 @@ clean:
 
 lint:
 	pre-commit run -a --hook-stage manual $(hook)
+	mypy kedro
```

Comment:
Is there any reason that we cannot keep this in pre-commit? For `ruff` we don't include this in `[test]` because pre-commit has its own virtual environment, could be useful to avoid dependencies problem.

Reply:
I did it because of this comment by @astrojuanlu https://github.com/kedro-org/kedro/issues/2156#issuecomment-1623648689

Reply:
My comment came from https://jaredkhan.com/blog/mypy-pre-commit

> We have seen many potential issues of running Mypy in Pre-commit:
> - Changing one file may cause a type error in another file, so we need to run Mypy on all files, not just those that have changed
> - We need to give Mypy access to the installed dependencies of the code it is type-checking, otherwise it can’t check the usages of those dependencies
> - Flags that reduce the strictness of Mypy like --ignore-missing-imports can give us false confidence

Reply:
In other words, pre-commit seems best suited to run linters that are stateless, have little configuration, and don't need the code to be installed. Neither of these applies to MyPy.